sympy/printing/latex.py

```diff
@@ -397,7 +398,21 @@ def convert(expr):
                 if self.order not in ('old', 'none'):
                     args = expr.as_ordered_factors()
                 else:
-                    args = expr.args
+                    args = list(expr.args)
+
+                # If quantities are present append them at the back
+                quantities_index = []
+                for i in range(len(args)):
+                    if isinstance(args[i], Quantity) or (isinstance(args[i], Pow) and
+                                                        isinstance(args[i].base, Quantity)):
+                        args.append(args[i])
+                        quantities_index.append(i)
+
+                if quantities_index != []:
+                    for i in range(len(args)-1, -1, -1):
+                        if quantities_index != [] and i == quantities_index[-1]:
+                            del quantities_index[-1]
+                            del args[i]
```

Comment:
You can do `sorted(args, key=lambda x: isinstance(x, Quantity) or (isinstance(x, Pow) and isinstance(x.base, Quantity)))`.